src/modules/land_detector/MulticopterLandDetector.cpp

```diff
@@ -86,17 +86,19 @@ MulticopterLandDetector::MulticopterLandDetector()
 
 void MulticopterLandDetector::_update_topics()
 {
+	LandDetector::_update_topics();
+
 	_actuator_controls_sub.update(&_actuator_controls);
 	_battery_sub.update(&_battery_status);
-	_vehicle_acceleration_sub.update(&_vehicle_acceleration);
 	_vehicle_angular_velocity_sub.update(&_vehicle_angular_velocity);
 	_vehicle_control_mode_sub.update(&_vehicle_control_mode);
-	_vehicle_local_position_sub.update(&_vehicle_local_position);
 	_vehicle_local_position_setpoint_sub.update(&_vehicle_local_position_setpoint);
 }
 
-void MulticopterLandDetector::_update_params()
+void MulticopterLandDetector::_update_params(const bool force)
 {
+	LandDetector::_update_params(force);
+
 	_freefall_hysteresis.set_hysteresis_time_from(false, (hrt_abstime)(1e6f * _param_lndmc_ffall_ttri.get()));
 
 	param_get(_paramHandle.minThrottle, &_params.minThrottle);
```

Comment:
@mcsauder this is not correct, it re-loads the params every cycle now, as there is no check for `_parameter_update_sub` here. You can move the `_parameter_update_sub.updated()` check out of  `LandDetector::_update_params` around the `_update_params();` in `LandDetector::Run()` (and get rid of the `force` argument).
(Alternatively you can rename this method here to `void updateParams() override;`, and then make `LandDetector::_update_params` non-virtual)
There also needs to be a call to `updateParams()` in `LandDetector::_update_params`.